chore: generate rc commits spreadsheet

sethkfman opened the pull request:
## **Description**

This PR includes a script to generate the changes between two branches. It will create a `csv` that is use by QA during release testing. 

## **Related issues**

Fixes:

## **Manual testing steps**

1. Fetch the latest from the repo
2. Run command `./generate-rc-commits vx.x.x release/x.x.x`
3. Look for csv called commits.csv 

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**


## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

sethkfman committed:
added RC commit per team script

sethkfman committed:
update PR script

sethkfman committed:
merge main

sethkfman committed:
updated lockfile

sethkfman committed:
script improvements

sethkfman committed:
sync with main

cortisiko commented on scripts/generate-rc-commits.js:
```diff
@@ -0,0 +1,243 @@
+    'Brian Bergeron',
+  ],
+  'Extension UX': ['David Walsh', 'vthomas13', 'Nidhi Kumari', 'Victor Thomas'],
+  'Extension Platform': [
```
any reason why we need the extension platform team?

cortisiko commented on scripts/generate-rc-commits.js:
```diff
@@ -0,0 +1,243 @@
+    'Kate Johnson',
+    'Brian Bergeron',
+  ],
+  'Extension UX': ['David Walsh', 'vthomas13', 'Nidhi Kumari', 'Victor Thomas'],
```
Perhaps rename to Wallet UX?

cortisiko commented on scripts/generate-rc-commits.js:
```diff
@@ -0,0 +1,243 @@
+    'Daniel Rocha',
+    'Howard Braham',
+    'Kate Johnson',
+    'Brian Bergeron',
```
Add mike to this list as well. @plasmacorral

plasmacorral replied:
I think Brian is on the assets team

cortisiko commented on scripts/generate-rc-commits.js:
```diff
@@ -0,0 +1,243 @@
+  Swaps: ['Daniel', 'Davide Brocchetto'],
+  Devex: ['Thomas Huang', 'Alex Donesky', 'jiexi', 'Zachary Belford'],
+  Security: ['witmicko'],
+  SDK: ['abretonc7s'],
```
@christopherferreira9, @omridan159 @andreahaku as well for the SDK

cortisiko commented on scripts/generate-rc-commits.js:
```diff
@@ -0,0 +1,243 @@
+  Devex: ['Thomas Huang', 'Alex Donesky', 'jiexi', 'Zachary Belford'],
+  Security: ['witmicko'],
+  SDK: ['abretonc7s'],
+};
```
can we also have a hardware wallet group as well? @vivek-consensys @stanleyyconsensys should be part of this group

cortisiko commented on scripts/generate-rc-commits.js:
```diff
@@ -0,0 +1,243 @@
+    'Matthew Walsh',
+    'cryptotavares',
+    'Vinicius Stevam',
+    'Derek Brans',
```
add @sleepytanya to this list as well.

cortisiko reviewed (changes requested):
left a few comments

plasmacorral commented on scripts/generate-rc-commits.js:
```diff
@@ -0,0 +1,243 @@
+    'Gustavo Antunes',
+    'Monte Lai',
+    'Daniel Rocha',
+    'Howard Braham',
```
Howard is not a member of the accounts team

jpcloureiro commented:
Is there anyone in charge of keeping the `JSON mapping authors to teams` up to date?

sethkfman commented:
@jpcloureiro The delivery team will be updating this each time the run an analysis. It flags the unknown members and they can be added to the sheet.

sethkfman committed:
updated team member names

sethkfman committed:
address lint errors

sethkfman committed:
update team names

sethkfman committed:
updated SDK user

plasmacorral commented on scripts/generate-rc-commits.js:
```diff
@@ -0,0 +1,235 @@
+    'Monte Lai',
+    'Daniel Rocha',
+    'Kate Johnson',
+  ],
```
Ok, one more team member to add to accounts, please: Charly Chevalier

plasmacorral committed:
Update generate-rc-commits.js

Update accounts team members

plasmacorral reviewed (dismissed):
looks good to me, Accounts team members are accurate

gantunesr commented:
Why certain names show up as their actual names and other as their GH handle? Which one is the correct?

sethkfman committed:
Merge branch 'main' into chore/generate-rc-commits

sethkfman commented:
@gantunesr It is based on the users git config and what is provided in the commit logs. Eventually we will be modifying the script to use team names.

sethkfman commented:
@SocketSecurity ignore npm/simple-git@3.22.0

wachunei committed:
refactor(ramp): persist and format amount in build quote view (#9039)

sethkfman committed:
chore: Sentry reduce traceSample rate to 0.04 (#9078)

## **Description**

This PR reduces the Sentry Performance sample rate by 20% since we are
meeting our 30-day cap. There are enough events to root cause current
performance issues.

## **Related issues**

Fixes:

## **Manual testing steps**

NA - non-functional change

## **Screenshots/Recordings**

NA - non-functional change

### **Before**

NA - non-functional change

### **After**

NA - non-functional change

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

owencraston committed:
fix: check for preferences controller in 036 migration (#9080)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**


1. What is the reason for the change?
- @Cal-L discovered a missing test case in [his
PR](https://github.com/MetaMask/metamask-mobile/pull/8967) for my
migration. In order to perfectly handle the edge cases for this
migration we must also check if the Preferences controller state is
defined.
- In order to unblock the above PR we need to handle this edge case.
2. What is the improvement/solution?
- check if the preferences controller state is defined
- add tests for the error cases and error messages

## **Related issues**

Fixes: N/A

## **Manual testing steps**

- [x] the tests should all pass. 
- [x] this migration was already tested and QAd in this previous
[PR](https://github.com/pulls?page=2&q=is%3Apr+author%3Aowencraston+archived%3Afalse+is%3Aclosed)
## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<img width="1246" alt="Screenshot 2024-03-27 at 11 54 27 AM"
src="https://github.com/MetaMask/metamask-mobile/assets/22918444/f9a34068-8ab9-45fa-8f80-1b45eb6115d0">



## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

Cal-L committed:
fix: Check for missing identities on migration 36 (#9082)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

This PR adds a check for ensuring that identities are defined under
PreferencesController for migration 36.
More context here -
https://github.com/MetaMask/metamask-mobile/pull/9080

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

cortisiko committed:
test: clean up detox build/run scripts (#9069)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been filled
out, and PR status checks have passed at least once.
-->

## **Description**

The purpose of this PR is to simplify the build and run commands for
running tests locally.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

tommasini committed:
feat: Update controller utils to 5.0.2 (#9079)

## **Description**

Updated the patch accordingly, and introduced networkIds list hardcoded
for we keep the ens legacy functions working

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

ENS solving: 


https://github.com/MetaMask/metamask-mobile/assets/46944231/d9f8fe17-c3a7-4c2e-b96c-7d72f4267dcc


### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

cortisiko committed:
test: stabilize swaps e2e tests (#9094)

Cal-L committed:
chore: Consolidate and simplify isTest condition (#9038)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

These changes both simplifies and consolidates the `isTest` condition,
which is mainly used for automated testing

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Curtis David <Curtis.David7@gmail.com>

Cal-L committed:
chore: Handle async migrations & add migration tests (#8967)

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

This PR enables the ability to write both asynchronous and synchronous
migrations. Moving forward, we should not have to worry about resolving
the incoming states within migration themselves. This change wraps each
migration in an asynchronous function, which resolves the state before
sending it to the next migration. We also added unit tests for checking
migration functionality using `createMigrate`

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

The videos below shows that migrations are still functioning as
expected. There is no change in user experience. In addition to E2E
tests, the main QA test to check is testing upgrading from current
release (7.18.0) to this branch.

Video of upgrade from 7.17.0 -> This branch. Example: Shows migration 35
being applied, which sets the displayed currency to ETH

https://github.com/MetaMask/metamask-mobile/assets/10508597/ab21c8f1-de46-40b4-85a7-5ece5fbe4425

Same flow going from 7.15.0 -> This branch

https://github.com/MetaMask/metamask-mobile/assets/10508597/4d03ddea-2391-4372-b553-f79c9bd5300f



## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it
is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

github-actions[bot] committed:
7.19.0 (#8984)

This is the release candidate for version 7.19.0.

[Release Testing
Tracker](https://docs.google.com/spreadsheets/d/1tsoodlAlyvEUpkkcNcbZ4PM9HuC9cEM80RZeoVv5OCQ/edit#gid=2124987401&range=A1)

---------

Co-authored-by: metamaskbot <metamaskbot@users.noreply.github.com>
Co-authored-by: sethkfman <seth.kaufman@consensys.net>
Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: CW <chris.wilcox@consensys.net>
Co-authored-by: Xiaoming Wang <dawnseeker8@gmail.com>
Co-authored-by: Omri Dan <61094771+omridan159@users.noreply.github.com>
Co-authored-by: Jyoti Puri <jyotipuri@gmail.com>
Co-authored-by: Kylan Hurt <kylanhurt@users.noreply.github.com>
Co-authored-by: sethkfman <10342624+sethkfman@users.noreply.github.com>
Co-authored-by: SamuelSalas <samuel.salas.reyes@gmail.com>
Co-authored-by: tommasini <tommasini15@gmail.com>
Co-authored-by: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Co-authored-by: Kylan Hurt <kylan.hurt@gmail.com>
Co-authored-by: Curtis David <Curtis.David7@gmail.com>

sethkfman committed:
update requires

sethkfman committed:
added team updates and recfactor category`

sethkfman committed:
Merge branch 'main' into chore/generate-rc-commits

sethkfman commented:
replaced by #9425
